src/vs/workbench/contrib/terminalContrib/chatAgentTools/browser/tools/monitoring/outputMonitor.ts

```diff
@@ -60,72 +65,210 @@ export class OutputMonitor extends Disposable implements IOutputMonitor {
 	constructor(
 		private readonly _execution: IExecution,
 		private readonly _pollFn: ((execution: IExecution, token: CancellationToken, taskService: ITaskService) => Promise<IPollingResult | undefined>) | undefined,
+		invocationContext: IToolInvocationContext,
+		token: CancellationToken,
+		command: string,
 		@ILanguageModelsService private readonly _languageModelsService: ILanguageModelsService,
 		@ITaskService private readonly _taskService: ITaskService,
 		@IChatService private readonly _chatService: IChatService,
 		@IChatWidgetService private readonly _chatWidgetService: IChatWidgetService
 	) {
 		super();
+		this.startMonitoring(command, invocationContext, token);
 	}
 
 	async startMonitoring(
 		command: string,
-		invocationContext: any,
+		invocationContext: IToolInvocationContext,
 		token: CancellationToken
-	): Promise<IPollingResult & { pollDurationMs: number }> {
+	): Promise<void> {
+		// Ensure this method always runs truly async so listeners can attach
+		await Promise.resolve();
```

Comment:
`await timeout(0)` would be more clear here, I'm not sure how awaiting a resolved promise behaves exactly.